src/libraries/Microsoft.Extensions.Diagnostics/tests/ListenerSubscriptionTests.cs

```diff
@@ -0,0 +1,381 @@
+                fakeListener.OnPublished = (instrument) =>
+                {
+                    publishCalled++;
+                    Assert.Equal(factory, instrument.Meter.Scope); // Local
+                    publishedTcs.TrySetResult(instrument);
+                    return (true, null);
+                };
+                fakeListener.OnCompleted = (instrument, state) =>
+                {
+                    onCompletedCalled++;
+                    Assert.Equal(1, onCompletedCalled);
+                    completedTcs.TrySetResult(instrument);
+                };
+                fakeListener.OnMeasurement = (instrument, measurement, tags, state) =>
+                {
+                    measurements.Add((int)measurement);
+                };
+
+                var subscription = new ListenerSubscription(fakeListener);
+                Assert.Null(fakeListener.Source);
+
+                subscription.Initialize();
+                Assert.Same(subscription, fakeListener.Source);
+
+                // No rules yet, so we shouldn't get any notifications.
+                var meter = factory.Create("TestMeter");
+                var counter = meter.CreateCounter<int>("counter", "blip", "I count blips");
+                counter.Add(1);
+
+                Assert.False(publishedTcs.Task.IsCompleted);
+                Assert.Equal(0, measurements.Count);
+                Assert.Equal(0, publishCalled);
+
+                // Add a rule that matches the counter.
+                subscription.UpdateRules(new[] { new InstrumentRule("TestMeter", "counter", null, MeterScope.Local, enable: true) });
+
+                Assert.True(publishedTcs.Task.IsCompleted);
+                Assert.Equal(1, publishCalled);
+
+                counter.Add(2);
+                Assert.Equal(1, measurements.Count);
+                Assert.Equal(2, measurements[0]);
+
+                Assert.False(completedTcs.Task.IsCompleted);
+
+                // Disable
+                subscription.UpdateRules(new[] { new InstrumentRule("TestMeter", "counter", null, MeterScope.Local, enable: false) });
+
+                Assert.True(completedTcs.Task.IsCompleted);
+                counter.Add(3);
+                // Not received
+                Assert.Equal(1, measurements.Count);
+
+                // Re-enable
+                subscription.UpdateRules(new[] { new InstrumentRule("TestMeter", "counter", null, MeterScope.Local, enable: true) });
+                Assert.Equal(2, publishCalled);
+
+                counter.Add(4);
+                Assert.Equal(2, measurements.Count);
+                Assert.Equal(4, measurements[1]);
+
+            }).Dispose();
+        }
+
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        // [InlineData(null, null, null)] // RemoteExecutor can't handle nulls
+        [InlineData("", "", "")]
+        [InlineData("*", "", "")]
+        [InlineData("*.*", "", "")]
```

Comment:
Logger does not allow more than one * in the string, I think we should be consistent with that precedent.